tests/unit/jsonFieldExpressionParser.js

```diff
@@ -0,0 +1,131 @@
+  testParsing("col:a", ['col', 'a']);
+
+  // less basic random babbling of test cases
+  testParsing('123', ['123']);
+  testParsing('123:abc', ['123', 'abc']);
+  testParsing('123:[1].abc', ['123', 1, 'abc']);
+  testParsing('123:[1.2].abc', ['123', '1.2', 'abc']);
+  testParsing('col:[1.2][1].abc', ['col', '1.2', 1, 'abc']);
+  testParsing('col:["1.2"][1].abc', ['col', '1.2', 1, 'abc']);
+  testParsing("col:['1']", ['col', '1']);
+
+  // with different quotes
+  testParsing("col:['[1.2]'][1].abc", ['col', '[1.2]', 1, 'abc']);
+  testParsing("col:['1']", ['col', '1']);
+
+  // array reference having only quotes
+  testParsing("col:[']", ['col', "'"]);
+  testParsing("col:['']", ['col', "''"]);
+  testParsing("col:[''']", ['col', "'''"]);
+
+  testParsing('col:["]', ['col', '"']);
+  testParsing('col:[""]', ['col', '""']);
+  testParsing('col:["""]', ['col', '"""']);
+
+  // array reference having quotes and brackets
+  testParsing('col:field["nofa\'].il"]', ['col', 'field', "nofa'].il"]);
+  testParsing("col:field['nofa\"].il']", ['col', 'field', 'nofa"].il']);
+
+  // quotes in dotreference part
+  testParsing("col:I'mCool", ['col', "I'mCool"]);
+  testParsing('col:PleaseMindThe"Quote"', ['col', 'PleaseMindThe"Quote"']);
+
+  // spaces in dotreference part
+  testParsing('col:I work too [100]', ['col', 'I work too ', 100]);
+
+  // new column reference style
+  testParsing('MyCupOfTeaTable.cupOfTea:I work too [100]', ['MyCupOfTeaTable.cupOfTea', 'I work too ', 100]);
+
+  // no column given
+  testFail(':[]');
+  testFail(':[nocolumn]');
+  testFail(':["nocolumn"]');
+  testFail(":['nocolumn']");
+  testFail(':nocolumn');
+
+  // invalid dotreference
+  testFail("col:[1].");
+
+  // trying to use index operator after dot
+  testFail("col:wat.[1]");
+  testFail("col:wat.['1']");
+  testFail('col:wat.["1"]');
+  testFail("col:wat[1].[1]");
+  testFail("col:wat[1].['1']");
+  testFail('col:wat[1].["1"]');
+
+  // opening square bracket in dot ref
+  testFail("col:a[1");
+  testFail("col:a['1'");
+  testFail('col:a["1"');
+  testFail("col:[1].a[1");
+  testFail("col:[1].a['1'");
+  testFail('col:[1].a["1"');
+
+  // closing square bracket in dot ref
+  testFail("col:a]1");
+  testFail("col:a]'1'");
+  testFail('col:a]"1"');
+  testFail("col:[1].a]1");
+  testFail("col:[1].a]'1'");
+  testFail('col:[1].a]"1"');
+
+  // invalid array references
+  testFail('col:wat[]');
+  testFail("col:wat[");
+  testFail("col:wat.a[");
+  testFail("col:wat]");
+  testFail("col:wat.a]");
+
+  testFail('col:wat[fa[il]');
+  testFail('col:wat[fa]il]');
+  testFail('col:wat.field[fa[il]');
+  testFail('col:wat.field[fa]il]');
+
+  // these should fail because inside quotes there is same type of
+  // quote => parser tries use stringWithoutSquareBrackets token
+  // for parsing => bracket in key fails parsing
+  testFail('col:field["fa"]il"]');
+  testFail("col:field['fa']il']");
+
+  describe("yanking up test coverage...", function () {
```

Comment:
well that was the truth :D since that parser is automatically generated and those features are never used in real code. should I change it to "field expression parser options" ?